.github/workflows/analysis.yaml

```diff
@@ -0,0 +1,46 @@
+name: cpp-linter
+
+on: [push, pull_request]
+defaults:
+  run:
+    shell: bash -e -l {0}
+jobs:
+  build:
+    runs-on: ubuntu-latest
+        
+    steps:
+    - name: Install LLVM and Clang
+      uses: egor-tensin/setup-clang@v1
+      with:
+        version: 18
+        platform: x64
+    
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Install Ninja
+      run: sudo apt install ninja-build -y
+
+    - name: Set conda environment
+      uses: mamba-org/setup-micromamba@main
+      with:
+        environment-name: myenv
+        environment-file: environment-dev.yml
+        init-shell: bash
+        cache-downloads: true
```

Comment:
Thinking again, I think we should only install `ninja` from conda-forge to use a single tool chain from the same prefix to prevent problem from occuring.

Since `ninja` is already part of the conda environment specification:

https://github.com/xtensor-stack/sparrow/blob/e423d1ffc35dd9cf3a277503deb6a253c59fb853/environment-dev.yml#L7

why is its installation from apt required, here?